Database/Entities/MessageStatusEntity.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.ComponentModel.DataAnnotations.Schema;
+
+namespace Database.Entities;
+
+public class MessageStatusEntity
+{
+	public MessageStatusEntity(int messageId, int receiverId, int status, DateTime timeStamp)
+	{
+		MessageId = messageId;
+		ReceiverId = receiverId;
+		Status = status;
+		TimeStamp = timeStamp;
+	}
+
+	public int MessageId { get; set; }
+	public int ReceiverId { get; set; }
+	public int Status { get; set; }
+	public DateTime TimeStamp { get; set; }
+
+	[ForeignKey(nameof(MessageId))]
```

Comment:
Why are you using both FluentApi and annotations? =)

Reply:
There is no way to create a composite PK with data annotations.